cdb2api/cdb2api.c

```diff
@@ -1557,7 +1557,7 @@ static int newsql_disconnect(cdb2_hndl_tp *hndl, SBUF2 *sb, int line)
     if ((hndl->firstresponse &&
          (!hndl->lastresponse ||
           (hndl->lastresponse->response_type != RESPONSE_TYPE__LAST_ROW))) ||
-        (!hndl->firstresponse)) {
+        (!hndl->firstresponse) || hndl->in_trans) {
```

Comment:
Mohit, you're right: the API change is not required in order to fix the bug. However, I'd prefer to keep the change to cover all the bases..